src/utils.js

```diff
@@ -96,6 +96,10 @@ function xhrRequest(url, callback, method){
       callback(error, error.message);
     }
   };
+  // triggered by network problems
+  xhr.onerror = function () {
+    callback({ status: 0, message: "", object: xhr }, "");
+  };
```

Comment:
No, this is to restore previous `onreadystatechange`-powered behavior; [`onload` doesn't get triggered by network failures, but `onreadystatechange` does](http://stackoverflow.com/a/19247992). You can trigger this scenario by calling `utils.xhrRequest` while offline.